File: day-102/leetcode.cpp
// Sort a linked list of 0s, 1s and 2s
// Remove duplicate element from sorted Linked List:
// Merge two sorted linked lists:

class Solution {
    public:
      // Function to sort a linked list of 0s, 1s and 2s.
      Node* segregate(Node* head) {
  
          // Add code here
          Node *temp=head;
          int count1=0,count2=0,count0=0;
          
          
          while(temp){
              
              if(temp->data==0){
                  count0++;
              }
              if(temp->data==1){
                  count1++;
              }
              if(temp->data==2){
                  count2++;
              }
              temp=temp->next;
          }
          
          temp=head;
          
          while(count0--){
              temp->data=0;
              temp=temp->next;
              
          }
          while(count1--){
              temp->data=1;
              temp=temp->next;
              
          }
          while(count2--){
              temp->data=2;
              temp=temp->next;
              
          }
          return head;
      }
};


class Solution {
    public:
      // Function to remove duplicates from sorted linked list.
      Node* removeDuplicates(Node* head) {
  
          // code here
          Node *curr=head->next,*pre=head;
          
          while(curr){
              if(curr->data==pre->data){
                  pre->next=curr->next;
                  delete curr;
                  curr=pre->next;
              }else{
                  pre=pre->next;
                  curr=curr->next;
              }
          }
          
          return head;
          
          
      }
};



class Solution {
    public:
      Node* sortedMerge(Node* head1, Node* head2) {
          // code here
          Node *head=new Node(0);
          Node *tail=head;
          
          
          while(head1 && head2){
             
              if(head1->data<=head2->data){
                  tail->next=head1;
                  head1=head1->next;
                 
              }
              else{
                  tail->next=head2;
                  head2=head2->next;
                 
              }
              
              tail=tail->next;
          }
          
          if (head1) tail->next = head1;
          else tail->next = head2;
  
          return head->next;
         
          
      }
};
  

File: day-108/leetcode.js
// Merge k Sorted Lists



/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode[]} lists
 * @return {ListNode}
 */

const merge = (head1, head2) => {
    let dummy = new ListNode(0);
    let tail = dummy;

    while (head1 && head2) {
        if (head1.val <= head2.val) {
            tail.next = head1;
            head1 = head1.next;
        } else {
            tail.next = head2;
            head2 = head2.next;
        }
        tail = tail.next;
    }

    if (head1) tail.next = head1;
    if (head2) tail.next = head2;

    return dummy.next;
}

var mergeKLists = function(lists) {

   if (lists.length === 0) return null;

    let head = lists[0];

    for (let k = 1; k < lists.length; k++) {
        head = merge(head, lists[k]);
    }

    return head;
    
  
};




/**
 * @param {number[][]} image
 * @return {number[][]}
 */
var flipAndInvertImage = function(image) {
   
    let res=[]

    for(let k=0;k<image.length;k++){
    res.push(image[k].reverse().map((res)=>res==0?1:0))
    }

    return res
};